clients/src/main/java/org/apache/kafka/common/utils/Utils.java

```diff
@@ -1271,4 +1274,34 @@ private static byte checkRange(final byte i) {
         }
         return map;
     }
+
+    /**
+     * Convert an ISO8601 based timestamp to an epoch value
+     * @param timestamp to be converted
+     * @return epoch value of a given timestamp
+     * @throws ParseException for timestamp that doesn't follow ISO8601 format
+     */
+    public static long getDateTime(String timestamp) throws ParseException {
+        final String[] timestampParts = timestamp.split("T");
+        if (timestampParts.length < 2) {
+            throw new ParseException("Error parsing timestamp. It does not contain a 'T' according to ISO8601 format", timestamp.length());
+        }
+
+        final String secondPart = timestampParts[1];
+        if (secondPart == null || secondPart.isEmpty()) {
+            throw new ParseException("Error parsing timestamp. Time part after 'T' is null or empty", timestamp.length());
+        }
+
+        if (!(secondPart.contains("+") || secondPart.contains("-") || secondPart.contains("Z"))) {
```

Comment:
I know that this is exiting logic that was just moved, but I am not sure if I understand how it works?

Reply:
@mjsax please let me know if ur asking anything specific i will try to explore that bt following is my understanding about this fxn : -

-call from (streamResetter/ commandgroup) to reset the offset based on epooch value given timestamp.
- check the format is correct if correct else throw exception. 
- append "Z" for UTC timezone if not there.
- parse and get the time.

Reply:
@mjsax request you to please review this again.

Thanks !!

Reply:
Hi @mjax request you to please review once.

Reply:
I was looking up the format in more detail and understand now what going on. The code does not seem to be ideal IMHO, but no need to change it in this PR.